jupyterhub/spawner.py

```diff
@@ -495,7 +495,7 @@ def get_args(self):
         """
         args = [
             '--user="%s"' % self.user.name,
-            '--base-url="%s"' % self.user.server.base_url,
```

Comment:
The `server` object should probably be passed directly to the Spawner, rather than accessing it through `user.server`, which no longer resolves to the right instance.